api/desecapi/serializers.py

```diff
@@ -1,45 +1,53 @@
 from rest_framework import serializers
-from desecapi.models import Domain, Donation, User, RRset
+from desecapi.models import Domain, Donation, User, RR, RRset
 from djoser import serializers as djoserSerializers
-import json
 
 
-class JSONSerializer(serializers.Field):
-    def to_representation(self, obj):
-        return json.loads(obj)
-
-    def to_internal_value(self, data):
-        return json.dumps(data)
-
-
-class RecordsSerializer(JSONSerializer):
-    def to_internal_value(self, records):
-        if isinstance(records, str) or not all(isinstance(record, str) for record in records):
-            msg = 'Incorrect type. Expected a list of strings'
-            raise serializers.ValidationError(msg)
-
-        # https://lists.isc.org/pipermail/bind-users/2008-April/070148.html
-        if not len(records) < 4092:
-            msg = 'Records too long. Must be less than 4092 characters, but was %d'
-            raise serializers.ValidationError(msg % len(records))
-
-        return super().to_internal_value(records)
+class RRSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = RR
+        fields = ('content',)
 
 
-class GenericRRsetSerializer(serializers.ModelSerializer):
+class RRsetSerializer(serializers.ModelSerializer):
+    domain = serializers.StringRelatedField()
     subname = serializers.CharField(allow_blank=True, required=False)
     type = serializers.CharField(required=False)
-    records = RecordsSerializer()
+    records = serializers.SerializerMethodField()
 
 
     class Meta:
         model = RRset
         fields = ('domain', 'subname', 'name', 'records', 'ttl', 'type',)
 
-
-class RRsetSerializer(GenericRRsetSerializer):
-    # The value of this field is set in RRsetList.perform_create()
-    domain = serializers.SlugRelatedField(read_only=True, slug_field='name')
+    def _inject_records_data(self, validated_data):
+        records_data = [{'content': x}
+                        for x in self.context['request'].data['records']]
+        rrs = RRSerializer(data=records_data, many=True, allow_empty=False)
```

Comment:
will this cause `CREATE` statements to the database?

Reply:
You probably mean `INSERT`. No, we are just feeding user input to the serializer to use the validation capabilites.